Tools/Pipeline/Templates/Effect.fx

```diff
@@ -0,0 +1,53 @@
+﻿
+#define DIRECTX // comment this line for compiling to OpenGL platforms
```

Comment:
You don't have to do that.  MGFX pre-defines some macros depending on the target platform.

https://github.com/mono/MonoGame/blob/develop/Tools/2MGFX/ShaderInfo.cs#L38

On DirectX platforms:  `HLSL`, `SM4`
On OpenGL platforms:  `GLSL`, `OPENGL`

Also `DEBUG` is defined if building the shader for debugging.

Now I will admit we probably should clean those up.  `DIRECTX` and `OPENGL` are better core macros than what we have.

Still you should be able to use `#if OPENGL` only and get the right changes in place.